includes/class-wc-stripe-customer.php

```diff
@@ -150,10 +150,12 @@ public function set_customer_data( $data ) {
 	/**
 	 * Generates the customer request, used for both creating and updating customers.
 	 *
-	 * @param  array $args Additional arguments (optional).
+	 * @param  array        $args  Additional arguments for the request (optional).
+	 * @param WC_Order|null $order The order object (optional). If provided, billing details will be retrieved from the order.
+	 *
 	 * @return array
 	 */
-	protected function generate_customer_request( $args = [] ) {
+	protected function generate_customer_request( $args = [], $order = null ) {
```

Comment:
Given that all the calling functions are public, I think we should extract `$order` from `$args` just to be extremely defensive against existing callers of the functions.
```suggestion
	protected function generate_customer_request( $args = [], $order = null ) {
		// The $order parameter was added in 10.2.0, so check for `$args['order'] for backwards compatibility.
		if ( null === $order && isset( $args['order'] ) && $args['order'] instanceof WC_Order ) {
			$order = $args['order'];
		}
		// Always ensure the old 'order' key is unset
		unset( $args['order'] );
```

Reply:
If we add this code, we should also add an explicit test case for it to make sure that direct calls to the relevant functions don't send the order data to Stripe _and_ use the correct data from the order.